File: backend/src/main/java/com/example/backend/entity/Payment.java
package com.example.backend.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;
import java.util.List;

@Entity
@Table(name = "payments")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Payment {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String paymentStatus = "Pending";

    private Double amount;

    private String transactionReference;

    private LocalDateTime paymentDate = LocalDateTime.now();

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;

    @ManyToOne
    @JoinColumn(name = "payment_format_id")
    private PaymentFormat paymentFormat;

    @OneToMany(mappedBy = "payment", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Transaction> transactions;

    @Column(nullable = false)
    private Boolean isActive = true;
}

File: frontend/src/app/shared/models/payment.ts
export interface Payment {
  id: number;
  userId: number; // eklendi (backend'de var)
  amount: number;
  transactionReference: string; // backend ismiyle uyumlu hale geldi
  status: 'SUCCESS' | 'FAILED' | 'PENDING'; // PENDING backend'de vardı
  paymentDate: Date; // backend: payment_date
  paymentFormatId: number;
}
